drivers/serial/uart_xlnx_uartlite.c

```diff
@@ -96,20 +96,36 @@ static inline void xlnx_uartlite_write_tx_fifo(const struct device *dev,
 
 static int xlnx_uartlite_poll_in(const struct device *dev, unsigned char *c)
 {
-	if (xlnx_uartlite_read_status(dev) & STAT_REG_RX_FIFO_VALID_DATA) {
+	uint32_t status;
+	int key;
+	int ret = -1;
+
+	key = irq_lock();
```

Comment:
This should be using a spinlock to support SMP cases.

Reply:
done